app/facades/foods_facade.rb

```diff
@@ -0,0 +1,16 @@
+class FoodsFacade
+  class << self
+    def relevant_dishes(ingredient)
+      create_dishes(FoodsService.get_dishes_including_ingredient(ingredient))
+    end
+
+    private
+
+    def create_dishes(dish_data)
+      total_found = dish_data[:totalHits]
+      dish_data[:foods].map do |dishy_data|
+        Food.new(dishy_data, total_found)
+      end
```

Comment:
this `total_found` doesn't really have much to do with the individual food objects, but it does have something to do with the original query. We definitely do want to get this sent back, and I like that you're trying to wrap it in with the objects you're already creating. Something you could look into is the following syntax: 

`total_found, foods= FoodFacade.your_method(params[:q])` 

You could put this in your controller, and then have `your_method` return a 2d array where the first element was the totalHits, and the second element was the collection of food objects you created.

Reply:
My thought around suggesting this is to get you thinking about a better way to format the two pieces of data we need to display to the view. The totalHits value is the total number of hits you have from the query that was made, and currently, you're saving that number of total hits to EVERY food you create -- it's the same number, and it's being stored on every Food object. Instead of doing that, you could send it back to your controller. 

```ruby
     total_found = dish_data[:totalHits]

     foods_collection = dish_data[:foods].map do |dishy_data|        
         Food.new(dishy_data)      
      end

     [total_found, foods_collection]
```

Reply:
When you return an array, you can assign it to variables by using the syntax I put in my first comment above.